### DISCUSSION TITLE
ESP32 TFT_eSPI problem to display JPG converted to HEX images

### DISCUSSION THREAD

**Discussion by VE2DX** [ImpulseAdventure/GUIslice#458] [Q&A] [answered]
Hello Gang,

I am trying to get a JPEG image to a display of an ESP32 Wrover based platform (The WT32-SC01)

on the M5Stack platform the command used is ;

`M5.Lcd.drawJpg(smeterTop, sizeof(smeterTop), 0, 0, 320, 20);`

but can't find the equivalent of this command in TFT_eSPI lib?

here is a partial example of the jpg image converted to HEX;

`#include <pgmspace.h>  // PROGMEM support header`

`PROGMEM const uint8_t smeterTop[]={`
`0xff, 0xd8, 0xff, 0xe0, 0x00, 0x10, 0x4a, 0x46, 0x49, 0x46, 0x00, 0x01, 0x01, 0x00, 0x00, 0x01, `
`0x00, 0x01, 0x00, 0x00, 0xff, 0xdb, 0x00, 0x43, 0x00, 0x03, 0x02, 0x02, 0x02, 0x02, 0x02, 0x03, `

**Comment by Pconti31** [comment, marked as answer]
@VE2DX generally you should ask in these questions in the TFT_eSPI GitHub forum not here unless you have an issue with GUIslice API.
That said the reason you haven't found how to do this is simple.  TFT_eSPI doesn't have calls for this.  

You need to useTFT_eSPI along with  TFT_eFEX library and a Jpeg decoder lib.  

Google GitHub TFT_eFEX and you will find it and it has examples. 

Note that GUIslice and the GUIslice Builder (my project)handles the details for you. My utility GUIslice_Image2C will convert jpegs to C arrays in a format GUIslice can use.

Paul--

**Comment by Pconti31** [comment]
Actually, as I think about it further there is a sample that is called TFT_flash_jpg.ino that only needs the Jpeg decoder.
Paul--
